### ISSUE TITLE
Bug: use-isnan doesn't report on `SequenceExpression`s 

### ISSUE THREAD

**Issue by StyleShit** [eslint/eslint#18058]
### Environment

Node version: v20.10.0
npm version: v10.2.3
Local ESLint version: v9.0.0-alpha.2 (Currently used)
Global ESLint version: Not found
Operating System: darwin 22.6.0

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint use-isnan: ["error", {"enforceForIndexOf": true}]*/

[1, NaN, 2].indexOf((1, NaN))
```

### What did you expect to happen?

The rule should report on this line

### What actually happened?

There is no report

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgdXNlLWlzbmFuOiBbXCJlcnJvclwiLCB7XCJlbmZvcmNlRm9ySW5kZXhPZlwiOiB0cnVlfV0qL1xuXG5bMSwgTmFOLCAyXS5pbmRleE9mKCgxLCBOYU4pKSIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
Confirmed as a bug, thanks for the report.
